### PR TITLE
[rum] script error

### PR DESCRIPTION
**Pull request by hdelaby**
### What does this PR do?

* Add documentation on "Script error." troubleshooting

### Motivation

* User feedback

### Preview

https://docs-staging.datadoghq.com/hdelaby/script-error/real_user_monitoring/browser/collecting_browser_errors/?tab=npm#fix-script-error

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by hdelaby** [comment]
Note: I'm not super satisfied with the layout in the 2-steps resolution section. Can we do something better?

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 1]
```diff
@@ -160,6 +160,26 @@ try {
 {{% /tab %}}
 {{< /tabs >}}
 
+## Troubleshooting
+
+### Fix "Script error."
+
+{{< img src="real_user_monitoring/browser/script-error.png" alt="Real User Monitoring script error example" style="width:75%;" >}}
+
+For security reasons, browsers hide details from errors triggered by cross-origin scripts ([see CORS][6] for more information). Some of the reasons why you are seeing this error are:
```
Comment (bcaudan) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:169]: For more information, [MDN explanation about the topic](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror#notes) could be better than redirecting to CORS doc

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 2]
```diff
@@ -160,6 +160,26 @@ try {
 {{% /tab %}}
 {{< /tabs >}}
 
+## Troubleshooting
+
+### Fix "Script error."
```
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:165]:
```suggestion
### Fixing script errors
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 3]
```diff
@@ -160,6 +160,26 @@ try {
...
+
+{{< img src="real_user_monitoring/browser/script-error.png" alt="Real User Monitoring script error example" style="width:75%;" >}}
```
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:167]:
```suggestion
{{< img src="real_user_monitoring/browser/script-error.png" alt="Real User Monitoring script error example" style="width:75%;" >}}
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md (hunk 1 above)
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:169]:
```suggestion
For security reasons, browsers hide details from errors triggered by cross-origin scripts. When this happens, the Error Details tab shows:
```
Error:
Script error.
```
For more information about cross-origin scripts and why details are hidden, see [CORS][6] and [this Note on Global Event Handlers][7]. Some possible reasons for this error include:
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 4]
```diff
@@ -160,6 +160,26 @@ try {
...
+
+For security reasons, browsers hide details from errors triggered by cross-origin scripts ([see CORS][6] for more information). Some of the reasons why you are seeing this error are:
+- Your JavaScript files are hosted on a different hostname (for instance, `example.com` includes assets from `static.example.com`)
+- Your website includes JavaScript libraries hosted on a CDN
+- Your website includes 3rd-party JavaScript libraries hosted on the provider's servers
```
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:172]:
```suggestion
- Your JavaScript files are hosted on a different hostname (for instance, `example.com` includes assets from `static.example.com`).
- Your website includes JavaScript libraries hosted on a CDN.
- Your website includes third-party JavaScript libraries hosted on the provider's servers.
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 5]
```diff
@@ -160,6 +160,26 @@ try {
...
+
+Get visibility into cross-origin scripts by following these two steps:
+1. Call JavaScript libraries with `crossorigin="anonymous"`
+
+With `crossorigin="anonymous"`, the request to fetch the script is performed securely: no sensitive data is forwarded via cookies or HTTP authentication.
```
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:177]:
```suggestion
    With `crossorigin="anonymous"`, the request to fetch the script is performed securely. No sensitive data is forwarded via cookies or HTTP authentication.
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 6]
```diff
@@ -160,6 +160,26 @@ try {
...
+
+{{< img src="real_user_monitoring/browser/script-error.png" alt="Real User Monitoring script error example" style="width:75%;" >}}
+
+For security reasons, browsers hide details from errors triggered by cross-origin scripts ([see CORS][6] for more information). Some of the reasons why you are seeing this error are:
+- Your JavaScript files are hosted on a different hostname (for instance, `example.com` includes assets from `static.example.com`)
+- Your website includes JavaScript libraries hosted on a CDN
+- Your website includes 3rd-party JavaScript libraries hosted on the provider's servers
+
+Get visibility into cross-origin scripts by following these two steps:
+1. Call JavaScript libraries with `crossorigin="anonymous"`
```
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:175]:
```suggestion
1. Call JavaScript libraries with `crossorigin="anonymous"`.
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 7]
```diff
@@ -160,6 +160,26 @@ try {
...
+
+With `crossorigin="anonymous"`, the request to fetch the script is performed securely: no sensitive data is forwarded via cookies or HTTP authentication.
+
+2. Configure the **Access-Control-Allow-Origin** HTTP header
```
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:179]:
```suggestion
2. Configure the `Access-Control-Allow-Origin` HTTP header.
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 8]
```diff
@@ -160,6 +160,26 @@ try {
...
+
+The most common value for this header is `Access-Control-Allow-Origin: *`. With this configuration, all origins can fetch the resource.
+Alternatively, you may want to restrict which origins can access your resources. For instance: `Access-Control-Allow-Origin: www.example.com`
```
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:182]:
```suggestion
    The most common value for this header is `Access-Control-Allow-Origin: *`, which allows all origins to fetch the resource. Instead, restrict which origins can access your resources by setting, for example `Access-Control-Allow-Origin: www.example.com`.
```

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md [hunk 9]
```diff
@@ -171,3 +191,4 @@ try {
 [3]: /real_user_monitoring/browser/advanced_configuration/
 [4]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error
 [5]: /real_user_monitoring/error_tracking
+[6]: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
```
Comment (kayayarai) [inline content/en/real_user_monitoring/browser/collecting_browser_errors.md:194]:
```suggestion
[6]: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
[7]: https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror#notes

```

**Review by kayayarai — CHANGES_REQUESTED** [review]
Edit comments, including fixing the Vale check and incorporating Bastien's comment.

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md (hunk 2 above)
Comment (hdelaby) [reply content/en/real_user_monitoring/browser/collecting_browser_errors.md:165]: "Script error." is a well-known JavaScript error. See this [blog article](https://blog.sentry.io/2016/05/17/what-is-script-error) for instance. Can we agree on `Fixing "Script error."`?

**Comment by hdelaby** [comment]
Hey @kayayarai something must have gone wrong in a suggestion so I applied what I think was your intention in my latest commit. Regarding the name of that section, I'll let you judge what is best and merge afterwards.

File: content/en/real_user_monitoring/browser/collecting_browser_errors.md (hunk 2 above)
Comment (kayayarai) [reply content/en/real_user_monitoring/browser/collecting_browser_errors.md:165]: Yes, `Fixing "Script error." is fine with me.

**Review by kayayarai — APPROVED** [review]

**Pull request merged**